doc/computing/randomness.rst

```diff
@@ -0,0 +1,315 @@
+.. Places parent toc into the sidebar
+
+:parenttoc: True
+
+.. _randomness:
+
+Controlling randomness: good practices and common pitfalls
+==========================================================
+
+Some scikit-learn objects are inherently random. These are usually estimators
+(e.g. :class:`~sklearn.ensemble.RandomForestClassifier`) and cross-validation
+splitters (e.g. :class:`~sklearn.model_selection.KFold`). The randomness of
+these objects is controlled via their `random_state` parameter, as described
+in the :term:`Glossary <random_state>`. This section expands on the glossary
+entry, and describes good practice and common pitfalls w.r.t. to this
+subtle parameter.
+
+The `random_state` parameter determines whether multiple calls to :term:`fit`
+(for estimators) or to :term:`split` (for cv splitters) will produce the same
+results, according to these rules:
+
+- If an int is passed, calling `fit` or `split` multiple times yields the
+  same results.
+- If `None` or a `RandomState` instance is passed: `fit` and `split` will
+  yield different results each time they are called, producing an event chain
+  of maximal entropy. `None` is the default value for all `random_state`
+  parameters.
+
+  .. note::
+      Since passing `random_state=None` is equivalent to passing the global
+      `RandomState` instance from `numpy`
+      (`random_state=np.random.mtrand._rand`), we will not explicitly mention
+      `None` here, but everything that applies to instances also applies to
+      using `None`.
+
+.. warning:: TL;DR
+
+    Unless you know what you are doing, we recommend to use integers as the
+    `random_state` parameter of estimators and cv splitters, as it is usually
+    the safest option and it allows you to isolate different sources of
+    variance in your analysis. Leaving the default (`None`) or using
+    `RandomState` instances is sometimes useful to assess to statistical
+    significance of your results, but can also have surprising effects.
+
+Getting reproducible results across multiple executions
+-------------------------------------------------------
+
+In order to obtain reproducible (i.e. constant) results across multiple
+program executions, the easiest way is to set a `rng` variable to an
+integer value at the top of your program, and pass it down to any object that
+accepts a `random_state` parameter::
```

Comment:
I found it slightly confusing for this variable to be named `rng` for a few reasons:

- At first, I thought it was short for "range", which seemed like a strange name for an integer.
- Then, I realized it stood for "random number generator", which also seemed strange since this integer is not doing any "generation". (However, it seems like a good variable name for a RandomState instance.)
- Finally, I found it distracting to keep track of whether `rng` referred to an integer or a RandomState instance throughout the rest of the page.

That is all to say that I would suggest using a different name for this particular variable (not sure what) when it refers to an integer, but keep using `rng` as the variable that refers to a RandomState instance.

However, you should definitely ignore my suggestion if using the name "rng" is already a widely adopted convention for this variable.

Reply:
thanks for the comment. Indeed I think `rng`  is  a pretty common name (it is at least used everywhere in our code base). I hope that explicitly mentioning the acronym at the first occurrence will help avoid confusion